### ISSUE TITLE
Symbolic shape inference for IF operator

### ISSUE THREAD

**Issue by yd2102** [microsoft/onnxruntime#6884]
**Describe the bug**
The symbolic_shape_infer.py script throws exception when input to IF operator is a tensor with single element

**Urgency**
None

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows
- ONNX Runtime installed from (source or binary): Binary
- ONNX Runtime version: 1.6.0
- Python version: 3.8
- Visual Studio version (if applicable): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**To Reproduce**
Run symbolic_shape_infer.py with input/output models as well as --auto_merge parameter.
The ONNX model cannot be attached here but the issue is obvious (see attached screenshot).

**Expected behavior**
The symbolic shape inference logic for IF operator should be able to handle tensor input containing a single element.

**Screenshots**
![image](https://user-images.githubusercontent.com/33105179/109883818-d47a6e80-7c30-11eb-823a-9023d3d41827.png)

**Additional context**
The model can be executed using ONNX runtime without error. Here's the error message from symbolic shape inference:

  File "python\onnx\lib\site-packages\onnxruntime\tools\symbolic_shape_infer.py", line 1297, in infer_shapes
    all_shapes_inferred = symbolic_shape_inference._infer_impl()
  File "python\onnx\lib\site-packages\onnxruntime\tools\symbolic_shape_infer.py", line 1164, in _infer_impl
    self.dispatcher_[node.op_type](node)
  File "python\onnx\lib\site-packages\onnxruntime\tools\symbolic_shape_infer.py", line 694, in _infer_If
    if cond > 0:
TypeError: '>' not supported between instances of 'list' and 'int'

**Comment by skottmckay** [comment]
Can you try making this change in _infer_If? 

Original:
```
        if cond is not None:
            if cond > 0:
```
New:
```
        if cond is not None:
            # ONNX allows cond to be a scalar (shape []) or list with one element (shape [1])
            if cond is list:
                assert len(list) == 1
                cond = cond[0]
                
            if cond > 0:
```

**Comment by yd2102** [comment]
Will "cond" potentially be a multi-dimensional tensor containing a single value? If yes, then the above solution isn't general enough to handle this.

**Comment by skottmckay** [comment]
No. There is only one condition variable used to decide if the 'then' or 'else' branch of the If is executed, so it can either be a value with no dims (shape == []) or a 1D tensor with one element (shape == []).

**Comment by skottmckay** [comment]
There's a helper that can handle an ndarray with a single value though - even if that shouldn't be possible in a valid model. the `if cond > 0:` in the original code could simply be replaced with `if as_scalar(cond) > 0:` with no other changes.

**Comment by yd2102** [comment]
Sorry for the late reply... @skottmckay There's one place that is missed: [here](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/symbolic_shape_infer.py#L767)

It should be:

if cond is not None and i_sub == (0 if **as_scalar**(cond) > 0 else 1):

**Comment by snnn** [comment]
This issue has been automatically closed as 'not planned' because it has been marked as 'stale' for more than 30 days without activity. If you believe this is still an issue, please feel free to reopen it.
